dtfs-central-api/src/v1/controllers/utilisation-report-service/post-upload-utilisation-report.controller/index.ts

```diff
@@ -73,8 +78,33 @@ export const postUploadUtilisationReport = async (req: PostUploadUtilisationRepo
           platform: 'PORTAL',
           userId: uploadedByUserId,
         },
+        transactionEntityManager,
       },
     });
+    await queryRunner.commitTransaction();
+    return updatedReport;
+  } catch (error) {
+    await queryRunner.rollbackTransaction();
+    if (error instanceof ApiError) {
+      throw error;
+    } else {
+      throw new TransactionFailedError();
+    }
```

Comment:
Will update the try-catch, the api error comes inside the eventHandler if it's an invalid event and can propagate up to here

Reply:
I forgot but the `TransactionFailedError` does extend `ApiError`, so it's just a specific case of `ApiError` with a message that looks similar to the one you put (and a 500). So I think you would only want to throw the `TransactionFailedError` rather than having this `if-else`